### PR TITLE
Fix issue where z_sendmany amount parameter is too strict

### PR DESCRIPTION
**Pull request by bitcartel**
Upstream treats an amount parameter of `1` the same as `1.0`.  Third-party would like this fixed.

### CODE REVIEW

**Comment by daira** [comment]
This has missed the cut for rc2.

**Comment by daira** [comment]
Oh, I didn't see this was a PR rather than a ticket. Reviewing.

**Comment by bitcartel** [comment]
Modified test so you can confirm.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3222,8 +3222,6 @@ Value z_sendmany(const Array& params, bool fHelp)
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, expected object");
         const Object& o = output.get_obj();
 
-        RPCTypeCheck(o, boost::assign::map_list_of("address", str_type)("amount", real_type));
```
Comment (daira) [inline src/wallet/rpcwallet.cpp]: Shouldn't this be replaced by a type check that accepts a real or integer. How does upstream type-check similar cases?

Comment (bitcartel) [inline src/wallet/rpcwallet.cpp]: sendtoaddress just uses `AmountFromValue` which we use later too.  The never use RPCTypeCheck.  sendmany doesn't use RPCTypeCheck either.  I added this check as part of z_sendmany, but it is too strict.

**Comment by daira** [comment]
ut(ACK+cov). @zkbot r+

**Comment by bitcartel** [comment]
@zkbot retry

**Comment by bitcartel** [comment]
See #1596

**Comment by daira** [comment]
@zkbot retry

**Pull request merged**